src/vs/platform/shell/node/shellEnv.ts

```diff
@@ -211,12 +225,69 @@ async function doResolveUnixShellEnv(logService: ILogService, token: Cancellatio
 				// https://github.com/microsoft/vscode/issues/22593#issuecomment-336050758
 				delete env['XDG_RUNTIME_DIR'];
 
-				logService.trace('getUnixShellEnvironment#result', env);
+				logService.trace('doResolveShellEnv#result', env);
 				resolve(env);
 			} catch (err) {
-				logService.error('getUnixShellEnvironment#errorCaught', toErrorMessage(err));
+				logService.error('doResolveShellEnv#errorCaught', toErrorMessage(err));
 				reject(err);
 			}
 		});
 	});
 }
+
+/**
+ * Returns powershell profile paths that are used to source its environment.
+ * This is used to determine whether we should resolve a powershell environment,
+ * potentially saving us from spawning a powershell process.
+ *
+ * @see https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-7.5
+ */
+async function getPowershellProfilePaths(psExecutable: string) {
+	const paths: string[] = [];
+	const userHome = homedir();
+	if (isWindows) {
+
+		// "The $PSHOME variable stores the installation directory for PowerShell" --
+		// but this is not set ambiently on the operating system.
+		let pshome = process.env.PSHOME;
+		if (!pshome) {
+			if (!isAbsolute(psExecutable)) {
+				const found = await findExecutable(psExecutable);
+				if (!found) {
+					return [];
+				}
+
+				pshome = dirname(found);
+			} else {
+				pshome = dirname(psExecutable);
+			}
+		}
+
+		paths.push(
+			join(pshome, 'Profile.ps1'), 													// All Users, All Hosts
```

Comment:
This list seems to miss the profile path that gets created by default when I run `if (-not (Test-Path $profile)) { New-Item $profile -Force }`. The location is: `C:\Users\bpasero\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1`